R/glm_models.R

```diff
@@ -1,19 +1,33 @@
-## * Generalized Linear Models R function
-## ** Conditional Probabilities
-.get_glm_condprob <- function(X, y, W,
-                              activation=function(x) x,
-                              log_prob=FALSE,
-                              ...) {
-    stop("not implemented yet")
+## * Generalized Linear Models R function 
+## ** Gradient functions
+## *** Gaussian
+## TODO(Ishmael): Add cost. gradient in math reform and reference
+##' @export
+.gaussian_loss <- function(X, y, w, lambda=0) {
+  innerProd <- X %*% w
+  losses <- 0.5 * (innerProd - y)^2
+  loss <- sum(losses)/NROW(X) + 0.5 * lambda * sum(w^2)
 }
-## ** Cost functions (negative log-likelihood)
-.get_glm_cost <- function(X, y, W, activation=function(x) x, ...) {
-    stop("not implemented yet")
+##' @export
+.gaussian_grad <- function(X, y, w, lambda=0) {
+  grads <- diag(c(X %*% w - y)) %*% X 
+  matrix(colMeans(grads) + lambda * sum(w), ncol=1)
 }
-## ** cost function gradient
-.get_glm_grad <- function(X, y, W, activation=function(x) x, ...) {
-    stop("not implemented yet")
+## *** Bernoulli
+##' @export
+.bernoulli_loss <- function(X, y, w, lambda=0) {
+  innerProd <- X  %*% w
+  losses <- log(1 + exp(-y * innerProd))
+  loss <- sum(losses)/NROW(X) + 0.5 * lambda * sum(w^2)
+  
+}
+##' @export
+.bernoulli_grad <- function(X, y, w, lambda=0) {
```

Comment:
Sure. I'll add this asap.

On Fri, Jul 24, 2015 at 10:41 AM, Toby Dylan Hocking <
notifications@github.com> wrote:

> In R/glm_models.R
> <https://github.com/IshmaelBelghazi/bigoptim/pull/14#discussion_r35428464>
> :
>
> >  }
> > -## ** cost function gradient
> > -.get_glm_grad <- function(X, y, W, activation=function(x) x, ...) {
> > -    stop("not implemented yet")
> > +## *** Bernoulli
> > +##' @export
> > +.bernoulli_loss <- function(X, y, w, lambda=0) {
> > +  innerProd <- X  %*% w
> > +  losses <- log(1 + exp(-y * innerProd))
> > +  loss <- sum(losses)/NROW(X) + 0.5 * lambda * sum(w^2)
> > +
> > +}
> > +##' @export
> > +.bernoulli_grad <- function(X, y, w, lambda=0) {
>
> Can you please add the loss and gradient equations in mathematical
> notation in comments here? And can you please add a reference to the
> equation number where the loss is defined in the SAG paper?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/IshmaelBelghazi/bigoptim/pull/14/files#r35428464>.
>